lockfile/src/spdx.rs

```diff
@@ -470,6 +507,98 @@ mod tests {
+            PackageVersion: 3.9.3
+            PrimaryPackagePurpose: LIBRARY
+            PackageSupplier: NOASSERTION
+            PackageDownloadLocation: https://pypi.org/project/aiohttp/3.9.3
+            FilesAnalyzed: false
+            PackageLicenseDeclared: NOASSERTION
+            PackageLicenseConcluded: Apache-2.0
+            PackageLicenseComments: <text>aiohttp declares Apache 2 which is not currently a valid SPDX License identifier or expression.</text>
+            PackageCopyrightText: NOASSERTION
+            PackageSummary: <text>Async http client/server framework (asyncio)</text>
+            ExternalRef: PACKAGE_MANAGER purl pkg:pypi/aiohttp@3.9.3
+            #####
+
+            PackageName: @colors/colors
+            SPDXID: SPDXRef-Package1
+            PackageVersion: 1.5.0
+            PackageDownloadLocation: http://github.com/DABH/colors.js.git
+            PackageSourceInfo: acquired package info from installed node module manifest file: /usr/local/lib/node_modules/npm/node_modules/@colors/colors/package.json
+            PackageOriginator: Person: DABH
+            PackageLicenseDeclared: MIT
+            PackageLicenseConcluded: MIT
+            PackageCopyrightText: NOASSERTION
+            PackageHomePage: https://github.com/DABH/colors.js
+            ExternalRef: SECURITY cpe23Type cpe:2.3:a:\@colors\/colors:\@colors\/colors:1.5.0:*:*:*:*:*:*:*
+            ExternalRef: SECURITY cpe23Type cpe:2.3:a:DABH:\@colors\/colors:1.5.0:*:*:*:*:*:*:*
+            ExternalRef: SECURITY cpe23Type cpe:2.3:a:dabh:\@colors\/colors:1.5.0:*:*:*:*:*:*:*
+            ExternalRef: PACKAGE-MANAGER purl pkg:npm/%40colors/colors@1.5.0
+
+            PackageName: @discoveryjs/json-ext
+            SPDXID: SPDXRef-Package2
+            PackageVersion: 0.5.6
+            PackageDownloadLocation: NOASSERTION
+            PackageSourceInfo: acquired package info from installed node module manifest file: /usr/local/go/src/cmd/vendor/github.com/google/pprof/third_party/d3flamegraph/package-lock.json
+            PackageLicenseDeclared: NONE
+            PackageLicenseConcluded: NONE
+            PackageCopyrightText: NOASSERTION
+            ExternalRef: SECURITY cpe23Type cpe:2.3:a:\@discoveryjs\/json-ext:\@discoveryjs\/json-ext:0.5.6:*:*:*:*:*:*:*
+            ExternalRef: SECURITY cpe23Type cpe:2.3:a:\@discoveryjs\/json-ext:\@discoveryjs\/json_ext:0.5.6:*:*:*:*:*:*:*
+            ExternalRef: SECURITY cpe23Type cpe:2.3:a:\@discoveryjs\/json_ext:\@discoveryjs\/json-ext:0.5.6:*:*:*:*:*:*:*
+            ExternalRef: SECURITY cpe23Type cpe:2.3:a:\@discoveryjs\/json_ext:\@discoveryjs\/json_ext:0.5.6:*:*:*:*:*:*:*
+            ExternalRef: SECURITY cpe23Type cpe:2.3:a:\@discoveryjs\/json:\@discoveryjs\/json-ext:0.5.6:*:*:*:*:*:*:*
+            ExternalRef: SECURITY cpe23Type cpe:2.3:a:\@discoveryjs\/json:\@discoveryjs\/json_ext:0.5.6:*:*:*:*:*:*:*
+            ExternalRef: PACKAGE-MANAGER purl pkg:npm/%40discoveryjs/json-ext@0.5.6
```

Comment:
It's due to having multiple packages being described which is probably not typical but `SPDX` does [support](https://github.com/spdx/spdx-spec/blob/a05c12a2dd4652b1396fd2659f2cd3ea1f37faba/schemas/spdx-schema.json#L218).

These packages `SPDXID`s are described via the now [deprecated](https://github.com/spdx/spdx-spec/blob/ecacc878b146c0bf38028397514615abb372e0e8/schemas/spdx-schema.json#L225) (or being deprecated) way found in `2.2`. 
```
DocumentDescribes: SPDXRef-Package1, SPDXRef-Package2
```

But this test also uses the `relationships` field.
```
Relationship: SPDXRef-DOCUMENT DESCRIBES SPDXRef-Package-1-cve-bin-tool
```

So that leaves `aiohttp` as the only package not described by this sbom.
Now I did put this together manually by copying chunks of other sobs, so the sbom document name probably doesn't make sense.

Reply:
> These packages `SPDXID`s are described via the now [deprecated](https://github.com/spdx/spdx-spec/blob/ecacc878b146c0bf38028397514615abb372e0e8/schemas/spdx-schema.json#L225) (or being deprecated) way found in `2.2`.
> 
> ```
> DocumentDescribes: SPDXRef-Package1, SPDXRef-Package2
> ```

I completely missed that line in this test file. Thanks for the clarification.